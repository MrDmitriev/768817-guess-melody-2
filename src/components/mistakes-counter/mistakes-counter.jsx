import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

export class MistakesCounter extends PureComponent {
  render() {
    const {mistakes} = this.props;
    return (
      <div className="game__mistakes">
        <div className={mistakes > 0 ? `wrong` : `correct`}></div>
        <div className={mistakes > 1 ? `wrong` : `correct`}></div>
        <div className={mistakes > 2 ? `wrong` : `correct`}></div>
      </div>
    );
  }
}

MistakesCounter.propTypes = {
  mistakes: PropTypes.number,
};

export default connect(
    (state) => ({
      mistakes: state.game.mistakes,
    })
)(MistakesCounter);
