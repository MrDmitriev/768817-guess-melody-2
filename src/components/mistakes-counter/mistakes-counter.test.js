import React from 'react';
import renderer from 'react-test-renderer';
import {MistakesCounter} from './mistakes-counter.jsx';

it(`should match snapshot`, () => {
  const props = {
    mistakes: 0,
  };

  const wrapper = renderer.create(<MistakesCounter {...props} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
