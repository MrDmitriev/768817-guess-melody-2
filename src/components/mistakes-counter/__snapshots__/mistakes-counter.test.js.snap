// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="correct"
  />
  <div
    className="correct"
  />
  <div
    className="correct"
  />
</div>
`;
